### PR TITLE
Add New Tool: StackExchange

### PR DESCRIPTION
**Pull request by richards199999**
# Description

I add another new tool for Dify - StackExchange. This tool enables the model to search for questions and answers on Stack Exchange and its sub-sites.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] The tool can be evoked correctly.
- [x] Since this tool doesn't require any authentication, there is no error related to this.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

---

The PR might contain the record of my previous PR. But I clean the files of my previous submitted tool. Only the new one.🤗

### CODE REVIEW

**Comment by crazywoola** [comment]
Please fix the lint error. :)

**Comment by richards199999** [comment]
I gonna cry =(

File: api/core/tools/provider/builtin/stackexchange/_assets/icon.png
Comment (crazywoola) [inline api/core/tools/provider/builtin/stackexchange/_assets/icon.png:1]: Please use an svg image instead.

Comment (richards199999) [reply api/core/tools/provider/builtin/stackexchange/_assets/icon.png:1]: oh sry i'll try again

**Comment by richards199999** [comment]
Why i always meet the same error in Tool Pytest? just can't anything wrong in my code.😭😭😭

```
FAILED api/tests/integration_tests/tools/test_all_provider.py::test_tool_providers - yaml.parser.ParserError: while parsing a block collection
in "<unicode string>", line 13, column 3:
  - name: intitle
    ^
expected <block end>, but found '?'
in "<unicode string>", line 55, column 3:
  llm_description: The Stack Excha ...
    ^
```

**Comment by richards199999** [comment]
Finally, everything fixed now. @crazywoola 
What a tough journey.😮‍💨😮‍💨😮‍💨

**Comment by richards199999** [comment]
I do have test it on my machine. here is the screenshot. @crazywoola 
image

and i have also update the yaml and py files.

**Review by crazywoola — APPROVED** [review]

**Pull request merged**